include/element.h

```diff
@@ -41,7 +41,7 @@ class Element {
   //! \param[in] particle_size Particle size
   //! \param[in] deformation_gradient Deformation gradient
   virtual Eigen::VectorXd shapefn(
-      const VectorDim& xi, const VectorDim& particle_size,
+      const VectorDim& xi, const unsigned& particle_size,
```

Comment:
Please leave these as `VectorDim`, material points can have different dimensions in different directions